qcsubmit/workflow_components/state_enumeration.py

```diff
@@ -169,30 +204,27 @@ def apply(self, molecules: List[Molecule]) -> ComponentResult:
             This is only possible using Openeye so far, if openeye is not available this step will fail.
         """
 
-        from openforcefield.utils.toolkits import OpenEyeToolkitWrapper
+        result = self._create_result(skip_unique_check=self.skip_unique_check)
 
-        result = self._create_result()
+        has_oe = self.cache
```

Comment:
Not sure if this is needed as the factory should raise an error if you add this component to a workflow but do not have openeye installed so we will never hit the apply method. The only time this would be needed is if the component is used in isolation but I am not sure how common that use case is.

Reply:
I am actually thinking that we should not actually make a hard requirement on OE, as that type of detail is an OpenFF TK implementation detail that we should not need to consider.

Reply:
I agree to some extent, the current method will raise an error when you create the workflow about openeye being missing before creating the dataset rather than running the whole workflow and finding all molecules have failed a component that has the missing dependency which might cause users to think there is a problem with the workflow rather than the environment.

Reply:
So in this case, we should be catching these types of environment things in either the constructors or the `add_workflow_component`. I agree that these errors would be confusing once the molecules are being processed.